Unable to build in Debug mode

Building a project using Pistache with CMake using `CMAKE_BUILD_TYPE=Debug` is currently impossible, since its linkage fails because of some unresolved symbols, similar to the ones described in #827. 
This, at least in my case, is caused by the `-fprofile-arcs` flag set in Pistache's CMakeLists.txt. As suggested by @Lectem in the mentioned issue, a clean CMake file would be ideal, but for the time being it could be useful to just remove the flag. 

Edit:
Compiler/linker used: Clang 11 with LLD, GCC 10.2.0 with ld.
GCC error log: https://del.dog/pistache-gcc-fprofile-arcs
Clang error log: https://del.dog/pistache-clang-fprofile-arcs

IIUC, "-fprofile-arcs" is present so that some CI tool would generate code-coverage graphs.  However, our travis-ci is currently broken (but will hopefully be fixed soon (issue #839).  

Looking at a section of linker error logs provided by Tachi107, I know what's going on (I've seen this many times in my own projects).  Pistache is built with profiling support enabled, but your own project is not linking in the profiler libs.  IMHO, Pistache's CMake should not enable profiling in the 'Debug' target, but should have an additional target called 'Profile' the should be used specifically for doing profiler-enabled builds.

I'm inclined to merge PR #841.  Then if we want "profile-arcs" for our CI, we implement isolated support for it in CMake.

```
[ 96%] Linking CXX executable Scraf
/usr/bin/ld: vendor/pistache/src/libpistache.a(http.cc.o): in function `_sub_I_00100_0':
/scraf-backend/vendor/pistache/src/common/http.cc:1009: undefined reference to `__gcov_init'
/usr/bin/ld: vendor/pistache/src/libpistache.a(http.cc.o): in function `_sub_D_00100_1':
/scraf-backend/vendor/pistache/src/common/http.cc:1009: undefined reference to `__gcov_exit'
/usr/bin/ld: vendor/pistache/src/libpistache.a(http.cc.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'

```

I still use hand-crafted, artisenal, Makefiles for my own project.  I use a hack like this to enable compile-time profiler support:
```
ifeq ($(COVERAGE),1)
  GCOV_CFLAGS:= --coverage
  GCOV_LDFLAGS:= --coverage
  O:=0
endif

ifeq ($(PROFILE),1)
  GPROF_CFLAGS:= -g -pg -no-pie       # For "gprof" and "valgrind"
endif

ifeq ($(SANITIZE),1)
  GPROF_CFLAGS:= -fsanitize=address
endif

```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -19,7 +19,7 @@ set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${PROJECT_SOURCE_DIR}/CMakeModules)
 include(CheckAtomic)
 
 if(CMAKE_BUILD_TYPE MATCHES "Debug")
-    set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -g -O0 -fprofile-arcs -ftest-coverage -fstack-protector-all --param=ssp-buffer-size=4")
+    set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -g -O0 -ftest-coverage -fstack-protector-all --param=ssp-buffer-size=4")
 endif()
 
 option(BUILD_SHARED_LIBS "build shared library" ON)

